pymatgen/transformations/advanced_transformations.py

```diff
@@ -1682,31 +1682,23 @@ def apply_transformation(self, structure: Structure) -> Structure:
         sc_not_found = True
 
         if self.force_diagonal:
-            # trans_mat_diagonal holds the diagonal of the trans_mat
-            trans_mat_diagonal = np.array([0, 0, 0])
-            trans_mat_diagonal_update = np.array([1, 1, 1])
-        else:
-            # target_threshold is used as the desired cubic side lengths
-            target_sc_size = self.min_length
+            scale = self.min_length / np.array(structure.lattice.abc)
+            self.transformation_matrix = np.diag(np.ceil(scale).astype(int))
+            st = SupercellTransformation(self.transformation_matrix)
+            return st.apply_transformation(structure)
 
+        # target_threshold is used as the desired cubic side lengths
+        target_sc_size = self.min_length
         while sc_not_found:
-            if self.force_diagonal:
-                # Update trans_mat (with diagonal constraint)
-                trans_mat_diagonal += trans_mat_diagonal_update
-                self.transformation_matrix = np.diag(trans_mat_diagonal)
-
-            else:
-                # Update trans_mat (without diagonal constraint)
-                target_sc_lat_vecs = np.eye(3, 3) * target_sc_size
-
-                self.transformation_matrix = (
-                        np.linalg.inv(lat_vecs) @ target_sc_lat_vecs
-                )
+            target_sc_lat_vecs = np.eye(3, 3) * target_sc_size
+            self.transformation_matrix = (
+                    np.linalg.inv(lat_vecs) @ target_sc_lat_vecs
```

Comment:
```suggestion
                     target_sc_lat_vecs @ np.linalg.inv(lat_vecs)
```

@utf I believe the order here might be wrong - sorry! Looking at `SupercellTransformation` within `pymatgen.transformations.standard_transformations` shows the relationship is new_lattice = transformation_matrix * old_lattice, which is also how it's written a few lines down with `proposed_sc_lat_vecs = self.transformation_matrix * lat_vecs`.